server.go

```diff
@@ -335,6 +337,35 @@ func (s *StubServer) initializeRouter() error {
 	return nil
 }
 
+func replaceRef(requestSchema *spec.Schema, componentSchemas map[string]*spec.Schema) *spec.Schema {
+	if requestSchema.Ref != "" {
+		return replaceRef(
+			componentSchemas[definitionFromJSONPointer(requestSchema.Ref)],
+			componentSchemas)
+	}
+
+	if requestSchema.Items != nil {
+		requestSchema.Items = replaceRef(requestSchema.Items, componentSchemas)
+		return requestSchema
+	}
+
+	if len(requestSchema.Properties) != 0 {
+		for prop, val := range requestSchema.Properties {
+			requestSchema.Properties[prop] = replaceRef(val, componentSchemas)
+		}
+		return requestSchema
+	}
+
+	if len(requestSchema.AnyOf) != 0 {
```

Comment:
This may be premature optimization, but it might be useful to provide a helper for `Schema` that allows a caller to iterate over the sub-schemas that will be found in it (just in case support for `oneOf` or something else is added in the future). Perhaps it could take a callback and allow the subschema to be modified by returning a non-`nil` value:

``` go
func replaceRef(requestSchema *spec.Schema, componentSchemas map[string]*spec.Schema) *spec.Schema {
    requestSchema.IterateSubSchemas(func(s *Schema) {
        return replaceRef(s, componentSchemas)
    })
    return requestSchema
}
```

Just an idea though! It's probably not necessary to do something like this right away.